spec/datadog/profiling/collectors/discrete_dynamic_sampler_spec.rb

```diff
@@ -15,10 +15,22 @@
     sampler
   end
 
+  def to_ns(secs)
+    (secs * 1e9).to_i
+  end
+
+  def to_sec(ns)
+    (ns / 1e9)
+  end
+
   def maybe_sample(sampling_seconds:)
-    start_ns = (@now * 1e9).to_i
-    end_ns = start_ns + (sampling_seconds * 1e9).to_i
-    sampler._native_after_sample(end_ns) / 1e9 if sampler._native_should_sample(start_ns)
+    sampled = false
+    if sampler._native_should_sample(to_ns(@now))
+      sampler._native_after_sample(to_ns(@now + sampling_seconds))
+      @now += sampling_seconds
+      sampled = true
+    end
+    sampled
```

Comment:
Minor: Tiny simplification ;)

```suggestion
    return false unless sampler._native_should_sample(to_ns(@now))
    
    sampler._native_after_sample(to_ns(@now + sampling_seconds))
    @now += sampling_seconds
    true
```